x/bank/keeper/keeper.go

```diff
@@ -87,14 +87,22 @@ func (k Keeper) SendCoins(ctx context.Context, fromAddr, toAddr sdk.AccAddress,
 		}
 	}
 
-	if err := k.bek.SendCoins(ctx, fromAddr, toAddr, evmCoins); err != nil {
-		return err
+	if evmCoins.Len() > 0 {
```

Comment:
It looks like the behavior is the same with or without this line, is it added for code efficiency and readability? Same as `cw20Coins.Len() > 0` and `cosmosCoins.Len() > 0`

Reply:
Never mind. I got it.

Reply:
This part is one of the biggest reasons for solving the problem. 
If this line were missing, []Coins{} would be passed, leaving Events unused. 
It's not that the existing amount couldn't be parsed; from Cosmos Bank's perspective, it's an empty Coins, so it printed an empty amount.